### PR TITLE
Caom24

### PR DESCRIPTION
**Pull request by hjeeves**

### CODE REVIEW

File: caom2-test-repo/src/main/java/ca/nrc/cadc/caom2/repo/integration/CaomRepoListTests23.java [hunk 1]
```diff
@@ -0,0 +1,510 @@
+/*
```
Comment (pdowler) [inline caom2-test-repo/src/main/java/ca/nrc/cadc/caom2/repo/integration/CaomRepoListTests23.java]: Listing did not change from 2.3 to 2.4; no need for this extra test class.

File: caom2-test-repo/src/main/java/ca/nrc/cadc/caom2/repo/integration/CaomRepoTupleTests23.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
+/*
```
Comment (pdowler) [inline caom2-test-repo/src/main/java/ca/nrc/cadc/caom2/repo/integration/CaomRepoTupleTests23.java]: read access tuple generation is fully tested on the 2.4 endpoint; no need for this extra test class.

File: caom2-repo-server/src/main/java/ca/nrc/cadc/caom2/repo/action/GetAction.java [hunk 3]
```diff
@@ -90,6 +90,7 @@
  *
  * @author pdowler
  */
+
```
[the remark this answers is not in the document]
Comment (hjeeves) [reply caom2-repo-server/src/main/java/ca/nrc/cadc/caom2/repo/action/GetAction.java]: ok

File: caom2-test-repo/src/main/java/ca/nrc/cadc/caom2/repo/integration/CaomRepoTupleTests23.java (hunk 2 above)
Comment (hjeeves) [reply caom2-test-repo/src/main/java/ca/nrc/cadc/caom2/repo/integration/CaomRepoTupleTests23.java]: ok

**Comment by hjeeves** [comment]
> There is a comment in task 12339 (which is marked complete in tinypm) that should modify PostAction (the comment in PutAction below relates to it) -- conditional update. What's up with that?

- got dropped, created 12588 so work can be done later

File: caom2-test-repo/src/main/java/ca/nrc/cadc/caom2/repo/integration/CaomRepoListTests23.java (hunk 1 above)
Comment (hjeeves) [reply caom2-test-repo/src/main/java/ca/nrc/cadc/caom2/repo/integration/CaomRepoListTests23.java]: ok

File: caom2-repo-server/src/main/java/ca/nrc/cadc/caom2/repo/action/PutAction.java [hunk 4]
```diff
@@ -94,6 +94,10 @@ public void doAction() throws Exception {
 
         checkWritePermission(uri);
 
+        // TODO: have a conditional check to see if the accMetaChecksum is different??
```
[the remark this answers is not in the document]
Comment (hjeeves) [reply caom2-repo-server/src/main/java/ca/nrc/cadc/caom2/repo/action/PutAction.java]: ok

File: caom2-repo-server/src/main/java/ca/nrc/cadc/caom2/repo/action/GetAction23.java [hunk 5]
```diff
@@ -87,30 +79,7 @@
...
-                writer.write(state.accMetaChecksum.toASCIIString());
-            } else {
-                writer.write("");
-            }
-            writer.endRecord();
-        }
-        writer.flush();
-        return bc.getByteCount();
+    protected ObservationWriter getObservationWriter() {
+        return new ObservationWriter("http://www.opencadc.org/caom2/xml/v2.3", false);
```
Comment (pdowler) [inline caom2-repo-server/src/main/java/ca/nrc/cadc/caom2/repo/action/GetAction23.java]: There is a contant for this: XmlContants.CAOM2_3_NAMESPACE

**Pull request merged**